### PR TITLE
[DNM] jewel: mon/OSDMonitor.cc: fix expected_num_objects interpret error

### PR DESCRIPTION
**Pull request by pdvian**
http://tracker.ceph.com/issues/23316

### CODE REVIEW

**Comment by pdvian** [comment]
@yangdongsheng @tchaikov @smithfarm backporting PR#19651 to jewel is not straight forward. Along with expected_num_objects changes, we need to skip mgr changes and qa/standalone/mon/osd-pool-create.sh related changes to be included in ./src/test/mon/osd-pool-create.sh. Looks like these changes are must in jewel, so tried backporting it.

**Comment by smithfarm** [comment]
retest this please

**Comment by smithfarm** [comment]
jenkins test docs

**Comment by smithfarm** [comment]
retest this please

**Comment by smithfarm** [comment]
Seems to be failing "make check"

FAIL: test/mon/osd-pool-create.sh

Error EINVAL: pool 'erasurecodes' cannot change to type replicated

**Comment by smithfarm** [comment]
retest this please

**Comment by smithfarm** [comment]
@pdvian The "make check" failure appears to be related to the changes in the PR. Could you take another look at the conflict resolution?

**Comment by pdvian** [comment]
@smithfarm Sure. Let me recheck this PR

File: src/mon/OSDMonitor.cc [hunk 1]
```diff
@@ -7010,8 +7010,9 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
     }
 
     bool implicit_ruleset_creation = false;
-    string ruleset_name;
-    cmd_getval(g_ceph_context, cmdmap, "ruleset", ruleset_name);
```
Comment (tchaikov) [inline src/mon/OSDMonitor.cc]: we don't need backport ruleset related changes.

**Comment by tchaikov** [comment]
closing in favor of #22050

**Pull request closed without merging**